File: code/matlab/Symbol.m
classdef Symbol
    %Symbol A linear combination of messages
    
    properties
        v_coeffs;
        val;
        msg_inds;
    end
    
    methods
        function obj = Symbol(v_coeffs, msg_vals, msg_inds)         
            obj.v_coeffs = v_coeffs;
            obj.val = dot(v_coeffs,msg_vals);
            obj.msg_inds = msg_inds;
        end
    end
    
end

File: code/tinyos/apps/TxIndexCoding/Symbol.py
import numpy as np

class Symbol:
	def __init__(self, v_coeffs, msg_vals, msg_inds):
		self.v_coeffs = np.copy(v_coeffs)
		self.val = np.dot(v_coeffs, msg_vals)
		self.msg_inds = np.copy(msg_inds)
